docs/entity_creation.md

```diff
@@ -0,0 +1,207 @@
+# Entity Creation
+
+
+
+Expanding the current selection of indexed entities is a way of increasing the coverage of queryable network interactions. 
+
+### Implementation
+
+
+#### Adding the entity schema
+
+in `schema.graphql`:
+
+The GraphQl schema is amended to include the new entity, with any fields that are needed to support new use cases. These attributes are preferably **only those required, in order to minimise storage requirements**.
+```
+0  type TestEntity @entity {
+1  		id: ID!
+2   	entity_field_a: String! @index
+3   	entity_field_b: String! @index
+4   	entity_field_c: String!
+5   	message: Message!
+6   	transaction: Transaction!
+7   	block: Block!
+8  }
+```
+
+- L0 - Entity is defined as a GraphQL type with the addition of the `@entity` annotation.
+
+- L1 - Entity requires a unique identifier, GraphQL type: ID - in SQL, this column will have the `text` type.
+
+- L2-4 - Needed entity attributes and their types are added, some with the `@index` tag to allow faster querying for anticipated use cases.
+
+- L5-8 - Entities: message, transaction and block are related.
+
+Updating the generated code to include these changes can be done by running the following: 
+```
+yarn codegen
+```
+
```

Comment:
Perhaps there's some low-effort change we could make which would allow us to move the custom ordering plugin code out of the subql submodule and into this repo. I think that would improve maintainability.